lib/core/src/sdk.rs

```diff
@@ -1629,53 +1630,71 @@ impl LiquidSdk {
                     }
                 );
 
-                let get_info_res = self.get_info().await?;
-                let validate_sufficient_funds = || {
-                    get_info_res.wallet_info.validate_sufficient_funds(
-                        self.config.network,
-                        receiver_amount_sat,
-                        *fees_sat,
-                        asset_id,
-                    )
-                };
-
-                let mut response = if asset_pay_fees {
-                    validate_sufficient_funds()?;
-                    self.pay_liquid_payjoin(liquid_address_data.clone(), receiver_amount_sat)
-                        .await?
-                } else {
-                    let fees_sat = fees_sat.ok_or(PaymentError::InsufficientFunds)?;
-
-                    let is_liquid_payment = *asset_id == self.config.lbtc_asset_id();
-                    let pay_with_bitcoin = match req.prepare_response.amount {
-                        Some(PayAmount::Asset {
-                            pay_with_bitcoin, ..
-                        }) => pay_with_bitcoin.unwrap_or(false),
-                        _ => false,
-                    };
+                let asset_pay_fees = req.use_asset_fees.unwrap_or_default();
+                let mut response = match amount.as_ref().is_some_and(|a| a.is_sideswap_payment()) {
```

Comment:
I think the entire `match` block that starts here could become a lot more readable if we extracted some logic into separate methods